sihl/src/database_migration_repo.ml

```diff
@@ -107,81 +119,48 @@ module MariaDb : Sig = struct
          version = VALUES(version),
          dirty = VALUES(dirty)
        |sql}
+         table)
   ;;
 
-  let upsert ~state =
+  let upsert table state =
     Database.query (fun (module Connection : Caqti_lwt.CONNECTION) ->
-        Connection.exec upsert_request (Migration.to_tuple state)
+        Connection.exec (upsert_request table) (Migration.to_tuple state)
         |> Lwt.map Database.raise_error)
   ;;
 end
 
 module PostgreSql : Sig = struct
   module Migration = Migration
 
-  let create_request =
+  let create_request table =
     Caqti_request.exec
       Caqti_type.unit
-      {sql|
-       CREATE TABLE IF NOT EXISTS core_migration_state (
+      (Format.sprintf
+         {sql|
+       CREATE TABLE IF NOT EXISTS %s (
          namespace VARCHAR(128) NOT NULL PRIMARY KEY,
          version INTEGER,
          dirty BOOL NOT NULL
        );
        |sql}
+         table)
   ;;
 
-  let create_table_if_not_exists () =
-    Database.query (fun (module Connection : Caqti_lwt.CONNECTION) ->
-        Connection.exec create_request () |> Lwt.map Database.raise_error)
-  ;;
-
-  let get_request =
-    Caqti_request.find_opt
-      Caqti_type.string
-      Caqti_type.(tup3 string int bool)
-      {sql|
-       SELECT
-         namespace,
-         version,
-         dirty
-       FROM core_migration_state
-       WHERE namespace = ?;
-       |sql}
-  ;;
-
-  let get ~namespace =
+  let create_table_if_not_exists table =
     Database.query (fun (module Connection : Caqti_lwt.CONNECTION) ->
-        Connection.find_opt get_request namespace
+        Connection.exec (create_request table) ()
         |> Lwt.map Database.raise_error)
-    |> Lwt.map (Option.map Migration.of_tuple)
   ;;
 
-  let get_all_request =
-    Caqti_request.collect
-      Caqti_type.unit
-      Caqti_type.(tup3 string int bool)
-      {sql|
-       SELECT
-         namespace,
-         version,
-         dirty
-       FROM core_migration_state;
-       |sql}
-  ;;
-
-  let get_all () =
-    Database.query (fun (module Connection : Caqti_lwt.CONNECTION) ->
-        Connection.collect_list get_all_request ()
-        |> Lwt.map Database.raise_error)
-    |> Lwt.map (List.map Migration.of_tuple)
-  ;;
+  let get = get
+  let get_all = get_all
 
-  let upsert_request =
+  let upsert_request table =
     Caqti_request.exec
       Caqti_type.(tup3 string int bool)
-      {sql|
-       INSERT INTO core_migration_state (
+      (Format.sprintf
+         {sql|
+       INSERT INTO %s (
```

Comment:
Could this be an attack vector for SQL injection?

Reply:
No, the table name is not based on any end user input. You need access to the env variables to set it.